### PR TITLE
add go mod support

### PR DESCRIPTION
**Pull request by Stebalien**
Also, do a bit of cleanup. Please read the commits separately.

### CODE REVIEW

**Comment by hsanjuan** [comment]
I did release cluster the other day with go-modules and without having to change anything in here... but other than that this seems ok.

File: build-go.sh [hunk 1]
```diff
@@ -156,7 +175,7 @@ function printInitialDistfile() {
 	jq ".name = \"$distname\"" |
 	jq ".owner  = \"`cat repo-owner`\"" |
 	jq ".description = \"`cat description`\"" |
-	jq ".date = \"`date '+%B %d, %Y'`\"" |
+	jq ".date = \"`date '+%b %d, %y'`\"" |
```
Comment (magik6k) [inline build-go.sh]: Why the lower case `%y`? I'd prefer `date '+%b %d, %Y'`

File: dists/fs-repo-migrations/Makefile [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
 all: dist
 
 dist: versions
-	../../build-go.sh fs-repo-migrations github.com/ipfs/fs-repo-migrations versions
+	../../build-go.sh ${distname} ${repo} ${package} versions
 	cp versions $(releases)/versions
 
 versions: filtered_versions
-	git ls-remote -t --refs $(repo) | egrep -o "refs/tags/v(.*)" | sed 's/refs\/tags\///' | sort > tag_versions
+	git ls-remote -t --refs https://$(repo) | egrep -o "refs/tags/v(.*)" | sed 's/refs\/tags\///' | sort > tag_versions
```
Comment (magik6k) [inline dists/fs-repo-migrations/Makefile]: We may also want to handle the `gx/vx.x.x` tags

Comment (magik6k) [reply dists/fs-repo-migrations/Makefile]: We also should export those common targets into a separate .mk file if possible

**Comment by Stebalien** [comment]
> I did release cluster the other day with go-modules and without having to change anything in here... but other than that this seems ok.

You ended up building with `go get` (`strings ipfs-cluster-ctl | grep 'distributions/.*/gopath/'`).

File: build-go.sh (hunk 1 above)
Comment (Stebalien) [reply build-go.sh]: Thanks for catching that! I probably just lowercased the line.

File: dists/fs-repo-migrations/Makefile (hunk 2 above)
Comment (Stebalien) [reply dists/fs-repo-migrations/Makefile]: We didn't rewrite the tags on go-ipfs, cluster, etc. (projects where we actually cut releases) so this shouldn't be an issue.

But yeah, we should have a common makefile.

Comment (Stebalien) [reply dists/fs-repo-migrations/Makefile]: Done.

**Comment by hsanjuan** [comment]
> > I did release cluster the other day with go-modules and without having to change anything in here... but other than that this seems ok.
> 
> You ended up building with `go get` (`strings ipfs-cluster-ctl | grep 'distributions/.*/gopath/'`).

:( always assumed it called `make build`

**Pull request merged**